examples/apiserver/apiserver.go

```diff
@@ -55,22 +55,48 @@ func newStorageFactory() genericapiserver.StorageFactory {
 	return storageFactory
 }
 
-func NewServerRunOptions() *genericoptions.ServerRunOptions {
-	serverOptions := genericoptions.NewServerRunOptions().WithEtcdOptions().WithSecureServingOptions().WithInsecureServingOptions()
-	serverOptions.InsecureServing.BindPort = InsecurePort
-	return serverOptions
+type ServerRunOptions struct {
+	GenericServerRunOptions *genericoptions.ServerRunOptions
+	Etcd                    *genericoptions.EtcdOptions
+	SecureServing           *genericoptions.SecureServingOptions
+	InsecureServing         *genericoptions.ServingOptions
```

Comment:
> seems worse to make every random API server enumerate these (and keep up with new options we want to be able to use consistently in all API servers)

In practice, this works out much better.  Not all API servers want the same thing.